include/nbl/video/surface/CSurfaceVulkan.h

```diff
@@ -0,0 +1,48 @@
+#ifndef __NBL_C_SURFACE_VULKAN_H_INCLUDED__
+#define __NBL_C_SURFACE_VULKAN_H_INCLUDED__
+
+#include "nbl/video/surface/ISurface.h"
+
+namespace nbl::video
+{
+
+class CVulkanConnection;
+class CVulkanPhysicalDevice;
+
+template <typename Window>
+class CSurfaceVulkan final : public CSurface<Window>
+{
+public:
+    using this_t = CSurfaceVulkan<Window>;
+    using base_t = CSurface<Window>;
+
+    static core::smart_refctd_ptr<this_t> create(
+        core::smart_refctd_ptr<video::CVulkanConnection>&& api,
+        core::smart_refctd_ptr<Window>&& window);
```

Comment:
having a non-inline method in a template class is not a good thing TM

Reply:
CC: @Erfan-Ahmadi

Reply:
caused a bit of circular dependancy but was able to do it by moving some functions around:
https://github.com/Devsh-Graphics-Programming/Nabla/commit/fe663184c0bc33c609e6936409621da77a9a7800

Reply:
done